### PR TITLE
Tween library

### PR DESCRIPTION
**Pull request by thegoldenmule**
Creating the PR now. Definitely a WIP.

### CODE REVIEW

**Comment by thegoldenmule** [comment]
@jsundra Opened this for visibility. Going to stub out some other tween types (vec3, col4) and then start on the JS API.

I wanted this to be a separate C# system, not just a JS interface. Also, I wanted a data object (`TweenData`) so that we can serialize these suckers if we want.

File: Assets/Source/Player/Scripting/Interface/TweenJs.cs [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+                        var callback = _onStart[i];
+                        _tween.OnStart += () => callback(context, new JsValue[0]);
+                    }
+
+                    // add completes
+                    for (int i = 0, len = _onComplete.Count; i < len; i++)
+                    {
+                        var callback = _onComplete[i];
+                        _tween.OnComplete += () => callback(context, new JsValue[0]);
+                    }
```
Comment (jsundra) [inline Assets/Source/Player/Scripting/Interface/TweenJs.cs]: Minor, but these lists could be nulled out after the callbacks are all added to the tween.

File: Assets/Source/Util/Tween/Tween.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        /// Called with an evaluated parameter.
+        /// </summary>
+        /// <param name="time">The time.</param>
+        protected abstract void Update(float time);
+    }
+
+    /// <summary>
+    /// Tween for a float.
+    /// </summary>
+    public class FloatTween : Tween
```
Comment (jsundra) [inline Assets/Source/Util/Tween/Tween.cs]: Might be good to break all the implementations into their own files as we add support for various different data types.

Comment (thegoldenmule) [reply Assets/Source/Util/Tween/Tween.cs]: For these small classes, I'd rather keep them all in one place.

File: Assets/Source/Player/Scripting/Interface/TweenJs.cs (hunk 1 above)
Comment (thegoldenmule) [reply Assets/Source/Player/Scripting/Interface/TweenJs.cs]: :+1:

**Comment by thegoldenmule** [comment]
Let's leave it off for now and see if we need it. We don't want to encourage an update loop!

**Pull request merged**